bundles/http/Citrix/Microsoft-Outlook/script.js

```diff
@@ -1,230 +1,1198 @@
+        throw new Error(`createOutlookOneTimeEventCurrentTimezone (${responseOfcreateOutlookOneTimeEventCurrentTimezone.status}: ${responseOfcreateOutlookOneTimeEventCurrentTimezone.statusText})`);
+    }
+    const user_id = [actionParameters.userId];
+    console.log(...user_id);
+    await calendarView(client, dataStore, user_id, startDate);
+    await myEvents(client, dataStore, user_id)
+}
+
+//service action for Create One Time Event with Custom Timezone
+async function createOutlookOneTimeEventCustomTimezone({
+    dataStore,
+    client,
+    actionParameters,
+}) {
+    const startDate = moment.utc().subtract(30, 'm').format();
+    const responseOfcreateOutlookOneTimeEventCustomTimezone = await client.fetch(
+        `/v1.0/me/events`,
+        {
+            method: "POST",
+            body: JSON.stringify({
+
+                subject: actionParameters.subject,
+                body: {
+                    contentType: "HTML",
+                    content: actionParameters.content
+                },
+                start: {
+                    dateTime: `${actionParameters.startDate}T${actionParameters.startTime}`,
+                    timeZone: actionParameters.timezone
+                },
+                end: {
+                    dateTime: `${actionParameters.endDate}T${actionParameters.endTime}`,
+                    timeZone: actionParameters.timezone
+                },
+                location: {
+                    displayName: actionParameters.location
+                },
+                attendees: [
+                    {
+                        emailAddress: {
+                            address: actionParameters.email1
+                        },
+                        type: actionParameters.type1
+                    },
+                    {
+                        emailAddress: {
+                            address: actionParameters.email2
+                        },
+                        type: actionParameters.type2
+                    }, {
+                        emailAddress: {
+                            address: actionParameters.email3
+                        },
+                        type: actionParameters.type3
+                    }, {
+                        emailAddress: {
+                            address: actionParameters.email4
+                        },
+                        type: actionParameters.type4
+                    },
+                    {
+                        emailAddress: {
+                            address: actionParameters.email5
+                        },
+                        type: actionParameters.type5
+                    }, {
+                        emailAddress: {
+                            address: actionParameters.email6
+                        },
+                        type: actionParameters.type6
+                    }
+                ],
+                allowNewTimeProposals: actionParameters.allowNewTimeProposals,
```

Comment:
This shouldn't be a variable, compare the service action with the Microsoft Outlook integration in the ecosystem.